### ISSUE TITLE
F-Droid build failed

### ISSUE THREAD

**Issue by licaon-kter** [Blockstream/green_android#173]
```
* What went wrong:
Execution failed for task ':lightning:compileReleaseJavaWithJavac'.
> Could not resolve all files for configuration ':lightning:releaseCompileClasspath'.
   > Could not find breez_sdk:bindings-android:0.1.0.
     Required by:
         project :lightning
```

No other maven except their own? https://sdk-doc.breez.technology/

Can we build this from source?

/LE: fyi https://gitlab.com/fdroid/fdroiddata/-/commit/554ea14f0cbc450dc5de881cb7ca077e3ac90f2d

**Comment by roeierez** [comment]
We are going to push to another public maven repo.

**Comment by licaon-kter** [comment]
@roeierez any timeframe? I see that 4.0.10 was tagged

**Comment by yaslama** [comment]
> ```
> * What went wrong:
> Execution failed for task ':lightning:compileReleaseJavaWithJavac'.
> > Could not resolve all files for configuration ':lightning:releaseCompileClasspath'.
>    > Could not find breez_sdk:bindings-android:0.1.0.
>      Required by:
>          project :lightning
> ```
How can we reproduce?
>
> No other maven except their own? https://sdk-doc.breez.technology/
Is this a problem for fdroid building?
> 
> Can we build this from source?
Yes. See https://github.com/breez/breez-sdk/tree/main/libs/sdk-bindings

**Comment by licaon-kter** [comment]
> How can we reproduce?

Remove breez.technology maven repo from code

> Is this a problem for fdroid building?

Yes, see https://f-droid.org/2022/07/22/maven-central.html and https://f-droid.org/en/docs/Inclusion_Policy/ from "Trusted maven repos"

**Comment by licaon-kter** [comment]
Ok, so... this `srclib/breez-sdk.yml` was added:
```
RepoType: git
Repo: https://github.com/breez/breez-sdk.git
Subdir: libs/sdk-bindings
```
and the recipe adapted:
```
  - versionName: 4.0.11
    versionCode: 22000411
    commit: 27bdf31e92b3d07acb2c912e6bcc141147b6eccf
    timeout: 25200
    subdir: green
    sudo:
      - apt-get update
      - apt-get install -y automake clang cmake jq libffi-dev libtool make pkg-config
        swig virtualenv protobuf-compiler
      - apt-get install -y openjdk-17-jdk-headless
      - update-java-alternatives -a
    gradle:
      - production
    srclibs:
      - breez-sdk@0.1.4
      - rustup@1.25.2
    rm:
      - bump_gdk.sh
    prebuild:
      - sed -i -e '/mvn.breez/d' ../build.gradle.kts
      - sed -i -e '/signingConfigs {/,+8d' -e '/signingConfigs.getByName/,+3d' -e
        '/versionNameSuffix/d' build.gradle.kts
      - sed -i -e '/dependencies {/aimplementation(files(\"libs/breez-sdk.aar\"))'
        build.gradle.kts
    build:
      - $$rustup$$/rustup-init.sh -y
      - source $HOME/.cargo/env
      - rustup install 1.68.0
      - export ANDROID_NDK_HOME=$$NDK$$
      - pushd $$breez-sdk$$
      - sed -i -e '/apple/d' makefile
      - make init
      - make bindings-android
      - popd
      - mkdir -p libs
      - cp $$breez-sdk$$/bindings-android/lib/build/outputs/aar/lib-release.aar libs/breez-sdk.aar
      - JAVA_HOME=$(readlink -f /usr/bin/javac | sed "s:/bin/javac::") PATH=$HOME/.local/bin:$PATH
        ../gdk/prepare_gdk_clang.sh arm64-v8a
    ndk: r25c
```

But fails:
```
...
> Task :lightning:compileReleaseJavaWithJavac FAILED
> Task :gdk:mergeReleaseJniLibFolders
> Task :no-gms:javaPreCompileRelease
> Task :green:dataBindingMergeDependencyArtifactsProductionRelease FAILED

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':lightning:compileReleaseJavaWithJavac'.
> Could not resolve all files for configuration ':lightning:releaseCompileClasspath'.
   > Could not find breez_sdk:bindings-android:0.1.4.
     Required by:
         project :lightning
```
full log: [com.greenaddress.greenbits_android_wallet_22000411.log.gz](https://github.com/Blockstream/green_android/files/12251705/com.greenaddress.greenbits_android_wallet_22000411.log.gz)

Surely this is an easy fix, right? Maybe I've not setup the kotlin thing to pickup the .aar correctly? Maybe easier?

**Comment by yaslama** [comment]
> > How can we reproduce?
> 
> Remove breez.technology maven repo from code

I want to run ``fdroid build`` in a vagrant vm to reproduce the problem but I cannot find the config.yml that you used to build.

**Comment by licaon-kter** [comment]
Save this as `metadata/com.greenaddress.greenbits_android_wallet.yml`: https://gist.github.com/licaon-kter/c86fd8a8de5b5536b5fea968f4d2f15b

run as: `fdroid build --verbose --scan-binary --server com.greenaddress.greenbits_android_wallet`

**Comment by angelix** [comment]
@licaon-kter is breez maven repo still an issue?

**Comment by licaon-kter** [comment]
@angelix yes, no progress here

**Comment by angelix** [comment]
Please check a260b4a46631da205784131c5671917fa35f065c for a temp solution.

In `common/build.gradle.kts` you have to make the following changes(comment/uncomment)

- comment breez kmp dependency
```
/**  --- Breez ------------------------------------------------------------------------------ */
// api(libs.breez.sdk.kmp)
/** ----------------------------------------------------------------------------------------- */
```

- uncomment  breez android dependency
```
// Temp fix for FDroid breez dependencies
api(libs.breez.sdk.android)
```

It it works i can add the change to `prepare_fdroid.sh` if it's used

**Comment by angelix** [comment]
@licaon-kter if you have any other ideas on how to solve this issue, please let me know.

**Comment by licaon-kter** [comment]
So updated recipe, afaik
```
  - versionName: 4.0.18
    versionCode: 22000418
    commit: 57dc1781c143d79dac5dc0b68fc5ec1d25aadbb1
    timeout: 25200
    subdir: green
    sudo:
      - apt-get update
      - apt-get install -y automake clang cmake jq libffi-dev libtool make pkg-config
        swig virtualenv xz-utils
      - apt-get install -y openjdk-17-jdk-headless
      - update-java-alternatives -a
    gradle:
      - production
    srclibs:
      - rustup@1.25.2
    prebuild:
      - sed -i -e '/mvn.breez/d' -e '/zendesk/d' ../build.gradle.kts
      - sed -i -e '/libs.breez.sdk.kmp/d' -e 's!// api(libs.breez.sdk.android)!api(libs.breez.sdk.android)!' ../common/build.gradle.kts
      - sed -i -e '/signingConfigs {/,+8d' -e '/signingConfigs.getByName/,+3d' -e
        '/versionNameSuffix/d' build.gradle.kts
    build:
      - $$rustup$$/rustup-init.sh -y
      - source $HOME/.cargo/env
      - rustup install 1.68.0
      - rustup target add aarch64-linux-android armv7-linux-androideabi
      - JAVA_HOME=$(readlink -f /usr/bin/javac | sed "s:/bin/javac::") PATH=$HOME/.local/bin:$PATH
        ../gdk/prepare_gdk_clang.sh arm64-v8a
      - JAVA_HOME=$(readlink -f /usr/bin/javac | sed "s:/bin/javac::") PATH=$HOME/.local/bin:$PATH
        ../gdk/prepare_gdk_clang.sh armeabi-v7a
    ndk: r25c
```

but fails somewhere: 
[com.greenaddress.greenbits_android_wallet_22000418.log.gz](https://github.com/Blockstream/green_android/files/13270271/com.greenaddress.greenbits_android_wallet_22000418.log.gz)

Ideas?

**Comment by angelix** [comment]
> So updated recipe, afaik
> 
> ```
>   - versionName: 4.0.18
>     versionCode: 22000418
>     commit: 57dc1781c143d79dac5dc0b68fc5ec1d25aadbb1
>     timeout: 25200
>     subdir: green
>     sudo:
>       - apt-get update
>       - apt-get install -y automake clang cmake jq libffi-dev libtool make pkg-config
>         swig virtualenv xz-utils
>       - apt-get install -y openjdk-17-jdk-headless
>       - update-java-alternatives -a
>     gradle:
>       - production
>     srclibs:
>       - rustup@1.25.2
>     prebuild:
>       - sed -i -e '/mvn.breez/d' -e '/zendesk/d' ../build.gradle.kts
>       - sed -i -e '/libs.breez.sdk.kmp/d' -e 's!// api(libs.breez.sdk.android)!api(libs.breez.sdk.android)!' ../common/build.gradle.kts
>       - sed -i -e '/signingConfigs {/,+8d' -e '/signingConfigs.getByName/,+3d' -e
>         '/versionNameSuffix/d' build.gradle.kts
>     build:
>       - $$rustup$$/rustup-init.sh -y
>       - source $HOME/.cargo/env
>       - rustup install 1.68.0
>       - rustup target add aarch64-linux-android armv7-linux-androideabi
>       - JAVA_HOME=$(readlink -f /usr/bin/javac | sed "s:/bin/javac::") PATH=$HOME/.local/bin:$PATH
>         ../gdk/prepare_gdk_clang.sh arm64-v8a
>       - JAVA_HOME=$(readlink -f /usr/bin/javac | sed "s:/bin/javac::") PATH=$HOME/.local/bin:$PATH
>         ../gdk/prepare_gdk_clang.sh armeabi-v7a
>     ndk: r25c
> ```
> 
> but fails somewhere: [com.greenaddress.greenbits_android_wallet_22000418.log.gz](https://github.com/Blockstream/green_android/files/13270271/com.greenaddress.greenbits_android_wallet_22000418.log.gz)
> 
> Ideas?

Seems for some reason gradle tries to download gdk binaries.

Can you comment out the following line in `/gdk/build.gradle.kts`?
```
commandLine("./fetch_android_binaries.sh")
```

**Comment by licaon-kter** [comment]
If I comment/remove that line if fails with
```
Execution failed for task ':gdk:fetchAndroidBinaries'.
> execCommand == null!
```

If I remove https://github.com/Blockstream/green_android/blob/release_4.0.18/gdk/build.gradle.kts#L25-L43 it fails later for some other reason :(

**Comment by licaon-kter** [comment]
```
> Task :green:mergeProductionReleaseJavaResource

> Task :green:minifyProductionReleaseWithR8 FAILED
ERROR: /home/vagrant/build/com.greenaddress.greenbits_android_wallet/gdk/build/intermediates/runtime_library_classes_jar/release/classes.jar: R8: Type com.blockstream.libgreenaddress.GDKJNI$NotificationHandler is defined multiple times: /home/vagrant/build/com.greenaddress.greenbits_android_wallet/gdk/build/intermediates/runtime_library_classes_jar/release/classes.jar:com/blockstream/libgreenaddress/GDKJNI$NotificationHandler.class, /home/vagrant/build/com.greenaddress.greenbits_android_wallet/green/build/intermediates/javac/productionRelease/classes/com/blockstream/libgreenaddress/GDKJNI$NotificationHandler.class

> Task :green:lintVitalAnalyzeProductionRelease

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':green:minifyProductionReleaseWithR8'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.R8Task$R8Runnable
   > Compilation failed to complete, origin: /home/vagrant/build/com.greenaddress.greenbits_android_wallet/gdk/build/intermediates/runtime_library_classes_jar/release/classes.jar:com/blockstream/libgreenaddress/GDKJNI$NotificationHandler.class
```

full log: @angelix  
[com.greenaddress.greenbits_android_wallet_22000418.log.gz](https://github.com/Blockstream/green_android/files/13560120/com.greenaddress.greenbits_android_wallet_22000418.log.gz)

**Comment by angelix** [comment]
> ```
> > Task :green:mergeProductionReleaseJavaResource
> 
> > Task :green:minifyProductionReleaseWithR8 FAILED
> ERROR: /home/vagrant/build/com.greenaddress.greenbits_android_wallet/gdk/build/intermediates/runtime_library_classes_jar/release/classes.jar: R8: Type com.blockstream.libgreenaddress.GDKJNI$NotificationHandler is defined multiple times: /home/vagrant/build/com.greenaddress.greenbits_android_wallet/gdk/build/intermediates/runtime_library_classes_jar/release/classes.jar:com/blockstream/libgreenaddress/GDKJNI$NotificationHandler.class, /home/vagrant/build/com.greenaddress.greenbits_android_wallet/green/build/intermediates/javac/productionRelease/classes/com/blockstream/libgreenaddress/GDKJNI$NotificationHandler.class
> 
> > Task :green:lintVitalAnalyzeProductionRelease
> 
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Execution failed for task ':green:minifyProductionReleaseWithR8'.
> > A failure occurred while executing com.android.build.gradle.internal.tasks.R8Task$R8Runnable
>    > Compilation failed to complete, origin: /home/vagrant/build/com.greenaddress.greenbits_android_wallet/gdk/build/intermediates/runtime_library_classes_jar/release/classes.jar:com/blockstream/libgreenaddress/GDKJNI$NotificationHandler.class
> ```
> 
> full log: @angelix [com.greenaddress.greenbits_android_wallet_22000418.log.gz](https://github.com/Blockstream/green_android/files/13560120/com.greenaddress.greenbits_android_wallet_22000418.log.gz)

Great, those logs gave me full understanding on what's going on. 

Seems that you both build and download gdk binaries.

If you still want to build gdk binaries from source, i'll get you a solution in a bit.

**Comment by licaon-kter** [comment]
Also
```
> Configure project :common

Please wait while Kotlin/Native compiler 1.9.20 is being installed.
Download https://download.jetbrains.com/kotlin/native/builds/releases/1.9.20/linux-x86_64/kotlin-native-prebuilt-linux-x86_64-1.9.20.tar.gz (198.16 MB)
Download kotlin-native-prebuilt-linux-x86_64-1.9.20.tar.gz finished, took 18 s 452 ms
Unpack Kotlin/Native compiler to /home/vagrant/.konan/kotlin-native-prebuilt-linux-x86_64-1.9.20
Unpack Kotlin/Native compiler to /home/vagrant/.konan/kotlin-native-prebuilt-linux-x86_64-1.9.20 finished, took 16 s 138 ms
```
...is a big no-no, ref: https://gitlab.com/fdroid/fdroiddata/-/issues/2989

**Comment by angelix** [comment]
> > Alternatively, you could try downloading the K/N distribution from maven central: https://central.sonatype.com/artifact/org.jetbrains.kotlin/kotlin-native-prebuilt
> 
> Would this be possible?

Using `kotlin.native.distribution.downloadFromMaven=true` i got the following. So i guess we are ok on that.

```
Please wait while Kotlin/Native compiler 1.9.20 is being installed.
Downloading https://repo.maven.apache.org/maven2/org/jetbrains/kotlin/kotlin-native-prebuilt/1.9.20/kotlin-native-prebuilt-1.9.20-macos-aarch64.tar.gz to /Users/angelos/.gradle/.tmp/gradle_download11606821103669116292bin
<==-----------> 20% CONFIGURING [15s]
```

**Comment by angelix** [comment]
> ```
> > Task :green:mergeProductionReleaseJavaResource
> 
> > Task :green:minifyProductionReleaseWithR8 FAILED
> ERROR: /home/vagrant/build/com.greenaddress.greenbits_android_wallet/gdk/build/intermediates/runtime_library_classes_jar/release/classes.jar: R8: Type com.blockstream.libgreenaddress.GDKJNI$NotificationHandler is defined multiple times: /home/vagrant/build/com.greenaddress.greenbits_android_wallet/gdk/build/intermediates/runtime_library_classes_jar/release/classes.jar:com/blockstream/libgreenaddress/GDKJNI$NotificationHandler.class, /home/vagrant/build/com.greenaddress.greenbits_android_wallet/green/build/intermediates/javac/productionRelease/classes/com/blockstream/libgreenaddress/GDKJNI$NotificationHandler.class
> 
> > Task :green:lintVitalAnalyzeProductionRelease
> 
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Execution failed for task ':green:minifyProductionReleaseWithR8'.
> > A failure occurred while executing com.android.build.gradle.internal.tasks.R8Task$R8Runnable
>    > Compilation failed to complete, origin: /home/vagrant/build/com.greenaddress.greenbits_android_wallet/gdk/build/intermediates/runtime_library_classes_jar/release/classes.jar:com/blockstream/libgreenaddress/GDKJNI$NotificationHandler.class
> ```

I think the issue is in the build recipe file.

`prepare_gdk_clang.sh` should be executed inside `gdk` folder.

Please try with this:
```
build:
      - $$rustup$$/rustup-init.sh -y
      - source $HOME/.cargo/env
      - rustup install 1.68.0
      - rustup target add aarch64-linux-android armv7-linux-androideabi
      - cd ../gdk
      - JAVA_HOME=$(readlink -f /usr/bin/javac | sed "s:/bin/javac::") PATH=$HOME/.local/bin:$PATH
        ./prepare_gdk_clang.sh arm64-v8a
      - JAVA_HOME=$(readlink -f /usr/bin/javac | sed "s:/bin/javac::") PATH=$HOME/.local/bin:$PATH
        ./prepare_gdk_clang.sh armeabi-v7a
```

**Comment by licaon-kter** [comment]
Thanks https://gitlab.com/fdroid/fdroiddata/-/commit/6fb621107e948d13d0c6c482a66daf7dc4e5879f

**Comment by licaon-kter** [comment]
new issue: https://gitlab.com/fdroid/fdroiddata/-/jobs/5807939772#L4517

**Comment by licaon-kter** [comment]
Another issue: https://gitlab.com/fdroid/fdroiddata/-/jobs/5898148283#L3553

**Comment by licaon-kter** [comment]
Another issue: https://gitlab.com/fdroid/fdroiddata/-/jobs/6123519265#L785 as https://github.com/Blockstream/green_android/blob/release_4.0.24/compose/build.gradle.kts#L76 is not-FOSS

**Comment by angelix** [comment]
> Another issue: https://gitlab.com/fdroid/fdroiddata/-/jobs/6123519265#L785 as https://github.com/Blockstream/green_android/blob/release_4.0.24/compose/build.gradle.kts#L76 is not-FOSS
> 
> /LE: fyi https://gitlab.com/fdroid/fdroiddata/-/commit/cd48f23c912f66b6298b57dd3e1108a925cb94d2

Will be removed on next version.

**Comment by licaon-kter** [comment]
Ping us when ready

**Comment by angelix** [comment]
@licaon-kter 
I added some metadata in #197. Can you confirm if the structure is ok? 
Thanks

**Comment by angelix** [comment]
@licaon-kter please check latest release

**Comment by licaon-kter** [comment]
> I added some metadata in https://github.com/Blockstream/green_android/pull/197. Can you confirm if the structure is ok?

looks ok

> please check latest release

will do asap

**Comment by licaon-kter** [comment]
Ummm
```
Please wait while Kotlin/Native compiler 1.9.22 is being installed.
Download kotlin-native-prebuilt-linux-x86_64-1.9.22.tar.gz finished, took 5 s 189 ms
```

better switch to Maven pls: https://central.sonatype.com/artifact/org.jetbrains.kotlin/kotlin-native-prebuilt ?

ref: https://gitlab.com/fdroid/fdroiddata/-/issues/2989

**Comment by licaon-kter** [comment]
I've added libc6-dev bit still complains about pthreads, what am I missing? 
[com.greenaddress.greenbits_android_wallet_22000426.log.gz](https://github.com/Blockstream/green_android/files/14725614/com.greenaddress.greenbits_android_wallet_22000426.log.gz)

**Comment by angelix** [comment]
> ```
> Download kotlin-native-prebuilt-linux-x86_64-1.9.22.tar.gz finished, took 5 s 189 ms
> ```

Already using `kotlin.native.distribution.downloadFromMaven=true` in [gradle.properties](https://github.com/Blockstream/green_android/blob/master/gradle.properties#L35)

**Comment by licaon-kter** [comment]
aaaand? :smile_cat:

**Comment by licaon-kter** [comment]
I've just started the build on my bullseye VM, brb

**Comment by licaon-kter** [comment]
Indeed, builds fine on Bullseye, still downloads kotlin binaries

**Comment by licaon-kter** [comment]
> Kotlin multiplatform binaries will always be downloaded as are not pre-packageed with Kotlin, but instead of being downloaded from jetbrains.com, are downloaded from public maven repository.

Ok, but... make it stop "Downloading from jetbrains"

**Comment by angelix** [comment]
From your logs
```
Please wait while Kotlin/Native compiler 1.9.22 is being installed.
Download kotlin-native-prebuilt-linux-x86_64-1.9.22.tar.gz finished, took 5 s 189 ms
Unpack Kotlin/Native compiler to /home/vagrant/.konan/kotlin-native-prebuilt-linux-x86_64-1.9.22
Unpack Kotlin/Native compiler to /home/vagrant/.konan/kotlin-native-prebuilt-linux-x86_64-1.9.22 finished, took 4 s 788 ms
```

From my machine with `--info`
```
Please wait while Kotlin/Native compiler 1.9.22 is being installed.
Downloading https://repo.maven.apache.org/maven2/org/jetbrains/kotlin/kotlin-native-prebuilt/1.9.22/kotlin-native-prebuilt-1.9.22-macos-aarch64.tar.gz to /Users/angelos/.gradle/.tmp/gradle_download1055476643572659025bin
```

Does it says `jetbrains.com` anywhere?

I think the `kotlin.native.distribution.downloadFromMaven=true` in `gradle.properties` solved that.

**Comment by licaon-kter** [comment]
Indeed, it might not, I don't have an actual URL in the logs.

Let me try something else, brb.

Any news on Bookworm?

**Comment by angelix** [comment]
> Any news on Bookworm?

I have some issues trying to make it run in my M2 processor.

Did you try installing `libevent-dev` ?

**Comment by licaon-kter** [comment]
I'm not sure which dep is needed (what hinted you about that one?), but I'll test asap

**Comment by angelix** [comment]
> I'm not sure which dep is needed (what hinted you about that one?), but I'll test asap

A message from gdk dev:
> libevent has some reported issues with android build, it has been fixed, but fixes have not been released in a new version yet. Though, I wonder why on our CI works and in theirs doesn't

**Comment by licaon-kter** [comment]
Who is "our" and who is "their"? :)

**Comment by angelix** [comment]
> Though, I wonder why on our **(Blockstream's)** CI works and in theirs**(F-Droid)** doesn't.

**Comment by licaon-kter** [comment]
So Blockstream (whatever that is?!) already builds in a Docker/VM with Bookworm?

Can you link to their setup?

**Comment by angelix** [comment]
I'll gather more info about that.

**Comment by angelix** [comment]
GDK needs to be patched to include libevent patches to work with newer cmake versions available in debian bookworm.

Relevant libevent MRs:
[1] https://github.com/libevent/libevent/pull/1353
[2] https://github.com/libevent/libevent/pull/1406

**Comment by licaon-kter** [comment]
@angelix how can we integrate these patches, at which step?

**Comment by angelix** [comment]
> @angelix how can we integrate these patches, at which step?

Better to wait for a new GDK/green_android release.

**Comment by angelix** [comment]
> @angelix how can we integrate these patches, at which step?

New version ready a14807e77150f00764f5579ccbe6c469d4257dd5

**Comment by angelix** [comment]
@licaon-kter kind reminder ^^

**Comment by licaon-kter** [comment]
was testing yes
```
...
> Task :gdk:generateReleaseLintModel FAILED
> Task :green:dataBindingMergeDependencyArtifactsProductionFDroidRelease
> Task :green:dataBindingMergeDependencyArtifactsProductionFDroidRelease FAILED
> Task :common:generateReleaseRFile FAILED

FAILURE: Build completed with 3 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':gdk:generateReleaseLintModel'.
> Dependency verification failed for configuration ':gdk:releaseCompileClasspath'
  One artifact failed verification: jna-5.14.0.pom (net.java.dev.jna:jna:5.14.0) from repository MavenRepo
  If the artifacts are trustworthy, you will need to update the gradle/verification-metadata.xml file. For more on how to do this, please refer to https://docs.gradle.org/8.7/userguide/dependency_verification.html#sec:troubleshooting-verification in the Gradle documentation.
```

what would I be missing? maybe it was a network glitch?

**Comment by angelix** [comment]
> was testing yes
> 
> ```
> ...
> > Task :gdk:generateReleaseLintModel FAILED
> > Task :green:dataBindingMergeDependencyArtifactsProductionFDroidRelease
> > Task :green:dataBindingMergeDependencyArtifactsProductionFDroidRelease FAILED
> > Task :common:generateReleaseRFile FAILED
> 
> FAILURE: Build completed with 3 failures.
> 
> 1: Task failed with an exception.
> -----------
> * What went wrong:
> Execution failed for task ':gdk:generateReleaseLintModel'.
> > Dependency verification failed for configuration ':gdk:releaseCompileClasspath'
>   One artifact failed verification: jna-5.14.0.pom (net.java.dev.jna:jna:5.14.0) from repository MavenRepo
>   If the artifacts are trustworthy, you will need to update the gradle/verification-metadata.xml file. For more on how to do this, please refer to https://docs.gradle.org/8.7/userguide/dependency_verification.html#sec:troubleshooting-verification in the Gradle documentation.
> ```
> 
> what would I be missing? maybe it was a network glitch?

@licaon-kter 
As a quick workaround remove the file `gradle/verification-metadata.xml`

**Comment by licaon-kter** [comment]
that does it, fyi https://gitlab.com/fdroid/fdroiddata/-/commit/8bb8ed02d993579a3c264c7ab33e1249ae4c52ed

**Comment by angelix** [comment]
@licaon-kter there is an issue in the current version in F-Droid related to dependency in [jitpack](https://github.com/jitpack/jitpack.io/issues/5752) stripping `jna` libs. 

There is a fix, but required a small change in the F-Droid build script doing changes in `common/build.gradle.kts`.

This is the new comments that needs to be changed.
```
            /**  --- Breez FDroid ----------------------------------------------------------------------- */
            // Temp fix for FDroid breez dependencies
            // api(libs.breez.sdk.android.get().toString()) { exclude(group = "net.java.dev.jna", module = "jna") }
            // implementation("${libs.jna.get()}@aar")
            /** ----------------------------------------------------------------------------------------- */
```

Maybe you find usefull the following `sed` commands.
```
sed -i -e 's/api(libs.breez.sdk.kmp)/\/\/ api(libs.breez.sdk.kmp)/g' common/build.gradle.kts
sed -i -e 's/\/\/ api(libs.breez.sdk.android/api(libs.breez.sdk.android/g' common/build.gradle.kts
sed -i -e 's/\/\/ implementation("${libs.jna/implementation("${libs.jna/g' common/build.gradle.kts
```


Also both anti-features are resolved by adding CountlyNoOp class to F-Droid flavor. Please remove the following flags if you see fit.
```
  NonFreeNet:
    en-US: Connects to countly.blockstream.com on app start
  Tracking:
    en-US: Sends analytics to countly.blockstream.com by default
```

**Comment by angelix** [comment]
@licaon-kter 
Please check a5fe83e903f580007dc5ec5316fa143be91aa6db. I'll tag it once you say it's ok to do so.

**Comment by licaon-kter** [comment]
> Also both anti-features are resolved by adding CountlyNoOp class to F-Droid flavor. 

how, where exactly?

**Comment by angelix** [comment]
> > Also both anti-features are resolved by adding CountlyNoOp class to F-Droid flavor.
> 
> how, where exactly?

Fixed in 9542a791d77a38a81260eab7eb353f19ec4eb323

**Comment by licaon-kter** [comment]
ah, you meant there were **already** added, not that get _"resolved by adding"_ something now :)

**Comment by angelix** [comment]
> ah, you meant there were **already** added, not that get _"resolved by adding"_ something now :)

Yes, my bad.

**Comment by licaon-kter** [comment]
@angelix builds fine, do you want the APK to test?

**Comment by angelix** [comment]
> @angelix builds fine, do you want the APK to test?

Νο, it's fine i guess, if there is a problem i'll need to create a new commit so...

**Comment by licaon-kter** [comment]
https://gitlab.com/fdroid/fdroiddata/-/commit/c7299a342f3c89a2fcfea9ac9611bf0cb83e7bd7 done

**Comment by licaon-kter** [comment]
ref: https://gitlab.com/fdroid/fdroiddata/-/jobs/7589564423#L3489

**Comment by angelix** [comment]
> ref: https://gitlab.com/fdroid/fdroiddata/-/jobs/7589564423#L3489
> 
> /LE: fyi https://gitlab.com/fdroid/fdroiddata/-/commit/9d2a478cd77c5034bc553ba2a7fbd359f64a44a8

@licaon-kter Should be fixed in 3dc7eed7b3dadc950465f6e79ab6dfe538e6b138 released as 4.0.33

**Comment by licaon-kter** [comment]
confirmed: https://gitlab.com/fdroid/fdroiddata/-/commit/639f3be11586ba9c8964155c6324889353d20127
